stock_view_adj_hls/models/stock_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from dateutil.relativedelta import relativedelta
+from odoo import fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    removal_date = fields.Date("Removal Date")
+
+    def _action_done(self):
+        res = super(StockMoveLine, self)._action_done()
+        for ml in self:
+            if ml.picking_id.picking_type_code == "incoming" and ml.removal_date:
+                removal_date = ml.removal_date + relativedelta(hours=3)
```

Comment:
Is that a better way to set this? Will the calculation be affected by the timezone of the user/server.

Reply:
Seems I tried to set the calculation best way but find the current easiest way to fix it. also, this thing will work based on user timezone.

Reply:
found one solution need be check, maybe it will work
`
local_date_to_dt = local_tz.localize(datetime.strptime(
 ts.date_to, DEFAULT_SERVER_DATE_FORMAT)).replace(
 hour=23, minute=59, second=59)`

Reply:
Hi, @TimLai125 Tested but not working. 
But Also get one idea.
Here we just only add Hours=3. It will convert time to based on user timezone like our mostly users based on Japan so it will use GMT+9. So the system will add  9 hours and on form view, the output will be 12:00.